### ISSUE TITLE
Comment query block isn't lazy loading meta data.

### ISSUE THREAD

**Issue by peterwilsoncc** [WordPress/gutenberg#40241]
### Description

In `build_comment_query_vars_from_block` the comment meta cache isn't updated on the basis the the comment data is lazy loaded. 

However, there are no calls within the code base to `wp_queue_comments_for_comment_meta_lazyload( array $comment_objects )`, so if any meta data is used in comments it will trigger a database call for each individual comment.

The function should be used each time a comments query is made using the build function:

```php
$comments = new WP_Comment_Query( build_comment_query_vars_from_block() );
wp_queue_comments_for_comment_meta_lazyload( $comments );
```

Options:
* Add all the calls
* Removing this line and come back to this in GB 14/WP 6.1 https://github.com/WordPress/gutenberg/blob/6f7c5b573a6023c2f2ce3eecd0d6b169ce4e964c/lib/compat/wordpress-6.0/blocks.php#L300
* If `WP_Comment_Query` is cached, make the query in `build_comment_query_vars_from_block()` so you can prime the meta data for lazy loading. It seems a little ugly though as building the arguments doens't nesesarily mean they will be used
* use `get_comments()` and propose a fix to Core that `get_comments()` triggers lazy loading of meta data by default. Add a coding standards sniff to Gutenberg that you call that rather than `WP_Comments_Query`.

### Step-by-step reproduction instructions

None, sorry, it's code.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by gziolo** [comment]
Closed with #40242. We still need to apply the same change in WordPress core. I'll take care of it shortly.

**Comment by gziolo** [comment]
I landed a temporary fix in the WordPress core with https://core.trac.wordpress.org/changeset/53214.

@peterwilsoncc, should we reopen the issue to work on a better approach for WP 6.1?
